packages/dds/tree/src/shared-tree/schematizedTree.ts

```diff
@@ -0,0 +1,124 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { unreachableCase } from "@fluidframework/common-utils";
+import {
+	schemaDataIsEmpty,
+	AllowedUpdateType,
+	Compatibility,
+	SimpleObservingDependent,
+} from "../core";
+import {
+	ViewSchema,
+	defaultSchemaPolicy,
+	ViewSchemaCollection,
+	ContextuallyTypedFieldData,
+} from "../feature-libraries";
+import { fail } from "../util";
+import { ISharedTreeBranch } from "./sharedTree";
+
+/**
+ * Takes in a tree and returns a view of it that conforms to the view schema.
+ * The returned view referees to and can edit the provided one: it is not a fork of it.
```

Comment:
When we actually start applying adapters, so the objects are not the same this could be confusing. One might assume this creates a new view/branch which is schematized. We might actually make the API do that in the future. The fact that we chose not to do that is an important part of the API since you must not merge both branches (the input and retuned one) currently.

How would a caller know they can't merge the branch they schematized and keep using the schematized version for example? How do they know they shouldn't? We need to make lifetime/ownership/mutation implications explicit in our API docs since this language doesn't handle that for us, and I don't think assuming some default contract is clear enough here.

Reply:
Eh, okay, I suppose a note about it being the same output tree as input tree might be helpful, but I still don't see why schematize has anything to do with forking. They're two orthogonal concepts and we should keep them that way. I don't think we should ever have this API fork automatically; the caller can easily fork their view before passing it in if they want. A comment like "this isn't a fork" just feels out of left field to me. That kind of comment increases our concept count and reduces the benefit we get from keeping our features orthogonal. I'd like a user to be able to use a SharedTree and all its schema features without ever even _thinking_ about forking. But with a comment like this, a user will read it and think, "Oh, I guess before I use schematize, I should go understand what forking is and how it works, because it is relevant here." And you're right that if it _is_ relevant, it does deserve a comment. But I still don't see any relationship, even lightly implied. In the absence of a comment, somebody who doesn't know about forking will assume the right thing, and I think that somebody who _does_ know about forking will also assume the right thing, because if this method did a fork internally it would be really misleading not to say so! And we don't make misleading docs here ;)

Anyway, I don't feel that strongly at this time; we can always revise comments later (and we'll be doing that anyway when we prepare for release; there are lots of things to cleanup!). Do whatever you think is best.